miso-service/src/main/java/uk/ac/bbsrc/tgac/miso/service/impl/DefaultLabService.java

```diff
@@ -75,10 +81,25 @@ public void update(Lab lab, Long instituteId) throws IOException {
   }
 
   @Override
-  public void delete(Long labId) throws IOException {
-    authorizationManager.throwIfNonAdmin();
-    Lab lab = get(labId);
-    labDao.deleteLab(lab);
+  public DeletionStore getDeletionStore() {
+    return deletionStore;
+  }
+
+  @Override
+  public AuthorizationManager getAuthorizationManager() {
+    return authorizationManager;
+  }
+
+  @Override
+  public ValidationResult validateDeletion(Lab object) {
+    ValidationResult result = new ValidationResult();
+
+    long usage = labDao.getUsage(object);
+    if (usage > 0L) {
+      result.addError(new ValidationError(usage + " sample" + (usage > 1L ? "s are" : " is") + " associated with this lab"));
```

Comment:
If someone is doing a bulk delete and this error is thrown, it's not obvious which lab was at fault (similar with institute above). Can this distinction be clarified?